CCPAPI/PAPI/ccPApi.cpp

```diff
@@ -0,0 +1,476 @@
+//#  it under the terms of the GNU Library General Public License as       #
+//#  published by the Free Software Foundation; version 2 or later of the  #
+//#  License.                                                              #
+//#                                                                        #
+//#  This program is distributed in the hope that it will be useful,       #
+//#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          #
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#          Copyright 2020 Paul RASCLE www.openfields.fr                  #
+//#                                                                        #
+//##########################################################################
+
+#include "ccPApi.h"
+
+#include <GeometricalAnalysisTools.h>
+
+//qCC_db
+#include <ccHObjectCaster.h>
+
+//libs/qCC_io
+#include<AsciiFilter.h>
+
+//system
+#include <unordered_set>
+
+//plugins
+#include "ccPluginInterface.h"
+#include "ccPluginManager.h"
+
+//qCC
+#include "ccCommon.h"
+
+#include "ccTrace.h"
+
+using namespace PAPI;
+
+// --- internal struct
+
+//* Extended file loading parameters, from plugins/ccCommandLineInterface.h
+struct CLLoadParameters: public FileIOFilter::LoadParameters
+{
+    CLLoadParameters() :
+            FileIOFilter::LoadParameters(), m_coordinatesShiftEnabled(false), m_coordinatesShift(0, 0, 0)
+    {
+        shiftHandlingMode = ccGlobalShiftManager::NO_DIALOG;
+        alwaysDisplayLoadDialog = false;
+        autoComputeNormals = false;
+        coordinatesShiftEnabled = &m_coordinatesShiftEnabled;
+        coordinatesShift = &m_coordinatesShift;
+    }
+
+    bool m_coordinatesShiftEnabled;
+    CCVector3d m_coordinatesShift;
+};
+
+//! internal attributes (cloned from plugins/ccCommandLineInterface.h)
+struct PAPI::ccPApi
+{
+    //! Currently opened point clouds and their filename
+    std::vector<CLCloudDesc> m_clouds;
+
+    //! Currently opened meshes and their filename
+    std::vector<CLMeshDesc> m_meshes;
+
+    //! Silent mode
+    bool m_silentMode;
+
+    //! Whether files should be automatically saved (after each process) or not
+    bool m_autoSaveMode;
+
+    //! Whether a timestamp should be automatically added to output files or not
+    bool m_addTimestamp;
+
+    //! Default numerical precision for ASCII output
+    int m_precision;
+
+    //! File loading parameters
+    CLLoadParameters m_loadingParameters;
+
+    //! Whether Global (coordinate) shift has already been defined
+    bool m_coordinatesShiftWasEnabled;
+
+    //! Global (coordinate) shift (if already defined)
+    CCVector3d m_formerCoordinatesShift;
+
+    //! Orphan entities
+    ccHObject m_orphans;
+};
+
+static PAPI::ccPApi* s_ccPApiInternals = nullptr;
```

Comment:
static member declarations usually happen at/near the top of the file.

In the case of the namespace change I mention above - right at the top of the PAPI namespace.